elephant/spectral.py

```diff
@@ -523,6 +529,352 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
+    if peak_resolution is not None:
+        if peak_resolution <= 0:
+            raise ValueError("peak_resolution must be positive")
+        nw = n_per_seg / fs * peak_resolution / 2
+        num_tapers = int(np.floor(2*nw) - 1)
+
+    if num_tapers is None:
+        num_tapers = int(np.floor(2*nw) - 1)
+    else:
+        if not isinstance(num_tapers, int):
+            raise TypeError("num_tapers must be integer")
+        if num_tapers <= 0:
+            raise ValueError("num_tapers must be positive")
+
+    # Generate frequencies of PSD estimate
+    if return_onesided:
+        freqs = np.fft.rfftfreq(n_per_seg, d=1/fs)
+    else:
+        freqs = np.fft.fftfreq(n_per_seg, d=1/fs)
+
+    # Zero-pad signal to fit segment length
+    remainder = length_signal % n_per_seg
+
+    data = np.pad(data, [(0, 0), (0, remainder)],
+                  mode='constant', constant_values=0)
+
+    # Generate array for storing cross spectra estimates of segments
+    cross_spec_estimates = np.zeros((n_segments,
+                                     data.shape[0],
+                                     data.shape[0],
+                                     len(freqs)),
+                                    dtype=np.complex64)
+    phase_spec_estimates = np.zeros((n_segments,
+                                     data.shape[0],
+                                     data.shape[0],
+                                     len(freqs)))
+
+    n_overlap_step = n_per_seg - n_overlap
+
+    for i in range(n_segments):
+        # Get slepian functions (sym='False' used for spectral analysis)
+        slepian_fcts = scipy.signal.windows.dpss(M=n_per_seg,
+                                                 NW=nw,
+                                                 Kmax=num_tapers,
+                                                 sym=False)
+
+        # Use broadcasting to match dime for point-wise multiplication
+        tapered_signal = (data[:, np.newaxis,
+                               i * n_overlap_step:
+                               i * n_overlap_step + n_per_seg]
+                          * slepian_fcts)
+
+        if return_onesided:
+            # Determine real Fourier transform of tapered signal
+            spectrum_estimates = np.fft.rfft(tapered_signal, axis=-1)
+
+        else:
+            # Determine full Fourier transform of tapered signal
+            spectrum_estimates = np.fft.fft(tapered_signal, axis=-1)
+
+        temp = (spectrum_estimates[np.newaxis, :, :, :]
+                * np.conjugate(spectrum_estimates[:, np.newaxis, :, :]))
+
+        # Average Fourier transform windowed signal
+        cross_spec_estimates[i] = np.mean(temp,
+                                          axis=-2,
+                                          dtype=np.complex64) / fs
+
+        phase_spec_estimates[i] = np.angle(np.mean(temp, axis=-2))
+
+    cross_spec = np.mean(cross_spec_estimates, axis=0)
+    phase_cross_spec = np.mean(phase_spec_estimates, axis=0)
```

Comment:
This geometric mean is not appropriate for averaging phase variables.

Reply:
`phase_cross_spec` was removed. Please see [df4a63b](https://github.com/NeuralEnsemble/elephant/pull/525/commits/df4a63bd80c55acbdaf1723de659e40421608991)